sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -130,6 +131,16 @@ final class DataFrameWriter private[sql](df: DataFrame) {
   }
 
   /**
+   * Specify the compression codec when saving it on hdfs
+   *
+   * @since 1.7.0
+   */
+  def compress(codec: Class[_ <: CompressionCodec]): DataFrameWriter = {
+    this.extraOptions += ("compression.codec" -> codec.getCanonicalName)
```

Comment:
Agree that we should not add interface for every configuration, but considering compression is a very common property, I feel it would be better to keep this interface. We also expose the compression api in RDD.saveAsXX, so think it would be better to be consistent here in dataframe